clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -386,17 +400,19 @@ object GarbageCollector {
     if (!storageNSForSdkClient.endsWith("/")) {
       storageNSForSdkClient += "/"
     }
+    
+    var removed : DataFrame = spark.emptyDataFrame
```

Comment:
More Scalaish (everything stays immutable):
```suggestion
    val removed = if (...) spark.emptyDataFrame else remove(...)
```

Reply:
Done